docs/docs/reference/unused-terms.md

```diff
@@ -0,0 +1,146 @@
+---
+layout: doc-page
+title: "Unused Parameters"
+---
+
+Why unused parameters?
+----------------------
+The following examples shows an implementation of a simple state machine which can be in a state `On` or `Off`.
+The machine can change state from `Off` to `On` with `turnedOn` only if it is currently `Off`. This last constraint is
+captured with the `IsOff[S]` implicit evidence which only exists for `IsOff[Off]`.
+For example, not allowing calling `turnedOn` on in an `On` state as we would require an evidence of type `IsOff[On]` that will not be found.
+
+```scala
+sealed trait State
+final class On extends State
+final class Off extends State
+
+@implicitNotFound("State is must be Off")
+class IsOff[S <: State]
+object IsOff {
+  implicit def isOff: IsOff[Off] = new IsOff[Off]
+}
+
+class Machine[S <: State] {
+  def turnedOn(implicit ev: IsOff[S]): Machine[On] = new Machine[On]
+}
+
+val m = new Machine[Off]
+m.turnedOn
+m.turnedOn.turnedOn // ERROR
+//                 ^
+//                  State is must be Off
+```
+
+These constraint that only depend on the types at the call site are completly resolved at compile time and never used at runtime.
```

Comment:
This phrase could be replaced with:

Note that in the code above the actual implicit arguments for `IsOff` are never used at runtime; they serve only to establish the right constraints at compile time.